src/DatumThese.ts

```diff
@@ -0,0 +1,496 @@
+* @since TODO
+*/
+export const successR = <E = never, A = never>(a: A): DatumThese<E, A> =>
+    D.refresh(right(a));
+
+/**
+* @since TODO
+*/
+export const failureR = <E = never, A = never>(e: E): DatumThese<E, A> =>
+    D.refresh(left(e));
+
+/**
+* @since TODO
+*/
+export const partialSuccessR = <E = never, A = never>(e: E, a: A): DatumThese<E, A> => 
+    D.refresh(both(e, a))
+
+/**
+ * @since TODO
+ */
+export function constInitial<E = never, D = never>(): DatumThese<E, D> {
+    return initial;
+}
+  
+  /**
+   * @since TODO
+   */
+export const constPending = <E = never, D = never>(): DatumThese<E, D> =>
+    pending;
+
+/**
+ * @since TODO
+ */
+export const isInitial = D.isInitial;
+
+/**
+ * @since TODO
+ */
+export const isPending = D.isPending;
+
+/**
+ * @since TODO
+ */
+export const isRefresh = D.isRefresh;
+
+/**
+ * @since TODO
+ */
+export const isReplete = D.isReplete;
+ 
+/**
+ * @since TODO
+ */
+export const isValued = D.isValued;
+ 
+/**
+ * @since TODO
+ */
+export const isRefreshLeft = <E, A>(
+    fea: DatumThese<E, A>
+): fea is D.Refresh<Left<E>> => isRefresh(fea) && isLeft(fea.value);
+  
+/**
+ * @since TODO
+ */
+export const isRefreshRight = <E, A>(
+    fea: DatumThese<E, A>
+): fea is D.Refresh<Right<A>> => isRefresh(fea) && isRight(fea.value);
+
+/**
+ * @since TODO
+ */
+export const isRefreshBoth = <E, A>(
+    fea: DatumThese<E, A>
+): fea is D.Refresh<Both<E, A>> => isRefresh(fea) && isBoth(fea.value)
+
+/**
+ * @since TODO
+ */
+export const isRepleteLeft = <E, A>(
+    fea: DatumThese<E, A>
+): fea is D.Replete<Left<E>> => isReplete(fea) && isLeft(fea.value);
+
+/**
+ * @since TODO
+ */
+export const isRepleteRight = <E, A>(
+    fea: DatumThese<E, A>
+): fea is D.Replete<Right<A>> => isReplete(fea) && isRight(fea.value);
+
+/**
+ * @since TODO
+ */
+export const isRepleteBoth = <E, A>(
+    fea: DatumThese<E, A>
+): fea is D.Replete<Both<E, A>> => isReplete(fea) && isBoth(fea.value)
+
+
+/**
+ * @since TODO
+ */
+export const isSuccess = <E, A>(fea: DatumThese<E, A>): fea is Success<A> =>
+    isValued(fea) && isRight(fea.value);
+
+/**
+* @since TODO
+*/
+export const isFailure = <E, A>(fea: DatumThese<E, A>): fea is Failure<E> =>
+    isValued(fea) && isLeft(fea.value);
+
+/**
+ * @since TODO
+ */
+export const isPartialSuccess = <E, A>(fea: DatumThese<E, A>): fea is PartialSuccess<E, A> =>
+    isValued(fea) && isBoth(fea.value);
+
+/**
+ * @since TODO
+ */
+export const toRefresh = <E, A>(fea: DatumThese<E, A>): DatumThese<E, A> =>
+    D.fold<These<E, A>, DatumThese<E, A>>(
+        constPending,
+        constPending,
+        constant(fea),
+        D.refresh
+    )(fea);
+
+/**
+ * @since TODO
+ */
+export const toReplete = <E, A>(fea: DatumThese<E, A>): DatumThese<E, A> =>
+    D.fold<These<E, A>, DatumThese<E, A>>(
+        constInitial,
+        constInitial,
+        D.replete,
+        constant(fea)
+)(fea);
+
+// TODO - fromThese vs fromThese2 ? Either equivalents are marked deprecated
```

Comment:
I did not implement the equivalent functionality from DatumThese, because the These equivalent of Either is deprecated